test/page.spec.js

```diff
@@ -46,58 +46,29 @@ module.exports.addTests = function(runner, expect, defaultBrowserOptions, puppet
       state.page = null;
     });
 
-    describe('Browser.version', function() {
-      it('should return whether we are in headless', async({browser}) => {
-        const version = await browser.version();
-        expect(version.length).toBeGreaterThan(0);
-        expect(version.startsWith('Headless')).toBe(headless);
-      });
-    });
-
-    describe('Browser.userAgent', function() {
-      it('should include WebKit', async({browser}) => {
-        const userAgent = await browser.userAgent();
-        expect(userAgent.length).toBeGreaterThan(0);
-        expect(userAgent).toContain('WebKit');
-      });
-    });
-
-    describe('Browser.process', function() {
-      it('should return child_process instance', async function({browser}) {
-        const process = await browser.process();
-        expect(process.pid).toBeGreaterThan(0);
-        const browserWSEndpoint = browser.wsEndpoint();
-        const remoteBrowser = await puppeteer.connect({browserWSEndpoint});
-        expect(remoteBrowser.process()).toBe(null);
-        await remoteBrowser.disconnect();
-      });
-    });
-
-    describe('Browser.Events.disconnected', function() {
-      it('should emitted when: browser gets closed, disconnected or underlying websocket gets closed', async() => {
-        const originalBrowser = await puppeteer.launch(defaultBrowserOptions);
-        const browserWSEndpoint = originalBrowser.wsEndpoint();
-        const remoteBrowser1 = await puppeteer.connect({browserWSEndpoint});
-        const remoteBrowser2 = await puppeteer.connect({browserWSEndpoint});
-
-        let disconnectedOriginal = 0;
-        let disconnectedRemote1 = 0;
-        let disconnectedRemote2 = 0;
-        originalBrowser.on('disconnected', () => ++disconnectedOriginal);
-        remoteBrowser1.on('disconnected', () => ++disconnectedRemote1);
-        remoteBrowser2.on('disconnected', () => ++disconnectedRemote2);
-
-        await remoteBrowser2.disconnect();
-        expect(disconnectedOriginal).toBe(0);
-        expect(disconnectedRemote1).toBe(0);
-        expect(disconnectedRemote2).toBe(1);
-
-        await originalBrowser.close();
-        expect(disconnectedOriginal).toBe(1);
-        expect(disconnectedRemote1).toBe(1);
-        expect(disconnectedRemote2).toBe(1);
-      });
-    });
+    const testFiles = [
+      'browser.spec.js',
+      'elementhandle.spec.js',
+      'jshandle.spec.js',
+      'tracing.spec.js',
+      'frame.spec.js',
+      'input.spec.js',
+      'network.spec.js'
+    ];
+
+    testFiles
+        .map(file => path.join(__dirname, file))
+        .forEach(file =>
+          require(file).addTests(
+              {
+                describe, xdescribe, fdescribe, it, fit, xit, beforeAll, beforeEach, afterAll, afterEach
+              },
+              expect,
+              defaultBrowserOptions,
```

Comment:
I think `defaultBrowserOptions` and `PROJECT_ROOT` are not needed for page subtests? Not sure what's the use of `puppeteer` as well.

Reply:
we do use some of them. I removed unnecessary params...